src/main/java/com/synopsys/integration/detect/ApplicationUpdater.java

```diff
@@ -272,11 +272,10 @@ private boolean runMainClass(Path jarPath)
             final Class<?> archiveClass = classMap.get("org.springframework.boot.loader.archive.Archive");
             final Class<?> mainClass = classMap.get("org.springframework.boot.loader.JarLauncher");
             final Constructor<?> jarLauncherConstructor = mainClass.getDeclaredConstructor(archiveClass);
-            jarLauncherConstructor.setAccessible(true);
-            final Object jarLauncher = jarLauncherConstructor.newInstance(jarFileArchive);
             final Class<?> launcherClass = 	classMap.get("org.springframework.boot.loader.Launcher");
             final Method launchMethod = launcherClass.getDeclaredMethod("launch", String[].class);
-            launchMethod.setAccessible(true);
```

Comment:
To invoke a Detect jar that is a Spring boot application, we need to modify the accessibility of the JarLauncher class and method as part of the Self-Update workflow. This is done only for the loaded class in memory and for the duration of the Detect scan. Sonar flags this modification but it is not an issue for Detect:
Only the JarLauncher class of Spring framework is modified from the loaded classes and not any of Detect or its dependency classes.